hnn_core/network.py

```diff
@@ -459,14 +469,34 @@ def get_spike_rates(self):
         cell_types = ['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']
         spike_rates = dict()
         all_spike_times = np.array(sum(self._times, []))
+        all_spike_types = np.array(sum(self._types, []))
+        all_spike_gids = np.array(sum(self._gids, []))
```

Comment:
something seems complicated with the logic. I can't quite pinpoint it without touching the code but it seems to me that you first flatten across trials and then do a for loop to access trial-specific info which makes it complicated.

Reply:
Since I reworked the rate calculation to start with individual gids, the flattening allows you to pre-allocate space for every gid. The rates are stored in the array `gid_spike_rate` with dimensions (trial x gid). This deals with the situation where specific gids don't spike on every trial; their inactivity should be included in the average across cells tyoes.

I'm sure there's a cleaner way to achieve this but that's the rationale.

Reply:
I'll take a look at this later. My brain doesn't want to think on a Saturday evening :) 

Do remember to update an example and also to update the "Attributes" section of the `Spikes` class once you add `tstart` and `tstop`

Reply:
No longer applicable to current commit.